Source/player.cpp

```diff
@@ -3269,520 +3270,341 @@ BOOL __fastcall PM_DoDeath(int pnum)
-
-	v1 = pnum;
-	p = pnum;
-	if ( (unsigned int)pnum >= MAX_PLRS )
-		TermMsg("CheckNewPath: illegal player %d", pnum);
-	v2 = v1;
-	if ( plr[v1].destAction == 20 )
-		MakePlrPath(v1, monster[plr[v2].destParam1]._mfutx, monster[plr[v2].destParam1]._mfuty, 0);
-	if ( plr[v2].destAction == 21 )
-		MakePlrPath(v1, plr[plr[v2].destParam1]._px, plr[plr[v2].destParam1]._py, 0);
-	if ( plr[v2].walkpath[0] == -1 )
-	{
-		v18 = plr[v2].destAction;
-		if ( v18 == -1 )
-			return;
-		v19 = plr[v2]._pmode;
-		if ( v19 == PM_STAND )
-		{
-			switch ( v18 )
-			{
-				case 9:
-					v20 = GetDirection(plr[v2].WorldX, plr[v2].WorldY, plr[v2].destParam1, plr[v2].destParam2);
-					goto LABEL_52;
-				case 10:
-					v30 = plr[v2].WorldY;
-					v31 = plr[v2].WorldX;
-					v32 = &plr[v2].destParam2;
-					v33 = &plr[v2].destParam1;
-					goto LABEL_59;
-				case 12:
-					v39 = GetDirection(plr[v2].WorldX, plr[v2].WorldY, plr[v2].destParam1, plr[v2].destParam2);
-					StartSpell(p, v39, plr[v2].destParam1, plr[v2].destParam2);
-					v40 = plr[v2].destParam3;
-					goto LABEL_66;
-				case 13:
-					v46 = plr[v2].destParam1;
-					arglista = v46;
-					v47 = v46;
-					v102 = abs(plr[v2].WorldX - object[v47]._ox);
-					v48 = abs(plr[v2].WorldY - object[v47]._oy);
-					if ( v48 > 1 )
-					{
-						v49 = object[v47]._oy;
-						if ( dObject[object[v47]._ox][v49-1] == -1 - arglista ) /* dungeon[39][112 * object[v47]._ox + 39 + v49] check */
-							v48 = abs(plr[v2].WorldY - v49 + 1);
-					}
-					if ( v102 > 1 || v48 > 1 )
-						break;
-					if ( object[v47]._oBreak != 1 )
-						goto LABEL_73;
-					goto LABEL_80;
-				case 14:
-					v50 = plr[v2].destParam1;
-					arglista = v50;
-					v47 = v50;
-					v103 = abs(plr[v2].WorldX - object[v47]._ox);
-					v51 = abs(plr[v2].WorldY - object[v47]._oy);
-					if ( v51 > 1 )
-					{
-						v52 = object[v47]._oy;
-						if ( dObject[object[v47]._ox][v52-1] == -1 - arglista ) /* dungeon[39][112 * object[v47]._ox + 39 + v52] check */
-							v51 = abs(plr[v2].WorldY - v52 + 1);
-					}
-					if ( v103 > 1 || v51 > 1 )
-						break;
-					if ( object[v47]._oBreak == 1 )
-					{
-LABEL_80:
-						v20 = GetDirection(plr[v2].WorldX, plr[v2].WorldY, object[v47]._ox, object[v47]._oy);
-LABEL_81:
-						v29 = p;
-LABEL_82:
-						StartAttack(v29, v20);
-					}
-					else
-					{
-						TryDisarm(p, arglista);
-LABEL_73:
-						OperateObject(p, arglista, 0);
+	/*
+	int oi;
+	*/
+	int i;
+	int x;
+	int y;
+	int d;
+	int xvel3;
+	int xvel;
+	int yvel;
+
+	if ( (DWORD)pnum >= MAX_PLRS ) {//8
+		TermMsg("CheckNewPath: illegal player %d", pnum); //12
+	}//10
+
+	if ( plr[pnum].destAction == 20 ) {//17-20
+		MakePlrPath(pnum, monster[plr[pnum].destParam1]._mfutx, monster[plr[pnum].destParam1]._mfuty, FALSE);//22-28
+	}//21
+
+	if ( plr[pnum].destAction == 21 ) {//29
+		MakePlrPath(pnum, plr[plr[pnum].destParam1]._px, plr[plr[pnum].destParam1]._py, FALSE);//31-37
+	}//30
+
+	if ( plr[pnum].walkpath[0] != -1) {//38

```

Comment:
WALK_NONE is currently not used anywhere, suggest changing its value to -1 and using it for places like this.

Reply:
I don't see this in your pr, where did you get the names from?

Reply:
Talking about _walk_path::WALK_NONE here.